ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/authentication/AuthenticationUtils.java

```diff
@@ -53,6 +53,10 @@ public static boolean isPermitted(
                   foundMatchingPermission.set(true);
                 }
               });
+          // exit if a matching permission was found, no need to keep checking
+          if (foundMatchingPermission.get()) {
```

Comment:
The more I read this code the more I think setting a flag async is not ideal...

Reply:
I don't love it either but it's not simple to change so it's not going to happen as part of this PR